MyRestaurant.Business.Tests/Repositories/TransactionRepositoryTest.cs

```diff
@@ -0,0 +1,187 @@
+﻿using FluentAssertions;
+using Moq;
+using MyRestaurant.Business.Dtos.V1;
+using MyRestaurant.Business.Errors;
+using MyRestaurant.Business.Repositories;
+using MyRestaurant.Business.Tests.Repositories.Fixtures;
+using MyRestaurant.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Net;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace MyRestaurant.Business.Tests.Repositories
+{
+    public class TransactionRepositoryTest : IClassFixture<TransactionRepositoryFixture>
+    {
+        private readonly TransactionRepositoryFixture _fixture;
+        public TransactionRepositoryTest(TransactionRepositoryFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        [Fact]
+        public async void GetTransactionsAsync_Returns_GetTransactionDtos()
+        {
+            //Arrange
+            _fixture.MockTransactionService.Setup(x => x.GetTransactionsAsync())
+                .ReturnsAsync(_fixture.Transactions);
+
+            var repository = new TransactionRepository(AutoMapperSingleton.Mapper, _fixture.MockTransactionService.Object);
+
+            //Act
+            var result = await repository.GetTransactionsAsync();
+
+            //Assert
+            var transactions = result.Should().BeAssignableTo<IEnumerable<GetTransactionDto>>().Subject;
```

Comment:
result.Should().BeOfType(typeof(IEnumerable<GetTransactionDto>)) can apply?

Reply:
cannot assign like this